control-plane-operator/controllers/hostedcontrolplane/pki/kas_test.go

```diff
@@ -0,0 +1,60 @@
+package pki
+
+import "testing"
+
+func Test_addBracketsIfIPv6(t *testing.T) {
+	tests := []struct {
+		name       string
+		apiAddress string
+		want       string
+		wantErr    bool
+	}{
+		{
+			name:       "given ipv4, it should not have brackets",
+			apiAddress: "192.168.1.1",
+			want:       "192.168.1.1",
+			wantErr:    false,
+		},
+		{
+			name:       "given an URL, it should not have brackets",
+			apiAddress: "https://test.tld:8451",
+			want:       "https://test.tld:8451",
+			wantErr:    false,
+		},
+		{
+			name:       "given an URL, it should not have brackets",
+			apiAddress: "test.tld:8451",
+			want:       "test.tld:8451",
+			wantErr:    false,
+		},
+		{
+			name:       "given simplified ipv6, it should return URL with brackets",
+			apiAddress: "fd00::1",
+			want:       "[fd00::1]",
+			wantErr:    false,
+		},
+		{
+			name:       "given an ipv6, it should return URL with brackets",
+			apiAddress: "fd00:0000:0000:0000:0000:0000:1:99",
+			want:       "[fd00:0000:0000:0000:0000:0000:1:99]",
+			wantErr:    false,
+		},
+		{
+			name:       "given wrong ipv6, it should return URL with brackets",
```

Comment:
did you mean "it should return error" here?

Reply:
yep xD, my bad